sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/WebJobsServiceBusTestBase.cs

```diff
@@ -0,0 +1,195 @@
+        protected const string FirstQueueNameKey = "%" + _firstQueueNameKey + "%";
+
+        private const string _secondQueueNameKey = "webjobstestqueue2";
+        protected const string SecondQueueNameKey = "%" + _secondQueueNameKey + "%";
+
+        private const string _thirdQueueNameKey = "webjobstestqueue3";
+        protected const string ThirdQueueNameKey = "%" + _thirdQueueNameKey + "%";
+
+        private const string _topicNameKey = "webjobstesttopic";
+        protected const string TopicNameKey = "%" + _topicNameKey + "%";
+
+        private const string _firstSubscriptionNameKey = "webjobstestsubscription1";
+        protected const string FirstSubscriptionNameKey = "%" + _firstSubscriptionNameKey + "%";
+
+        private const string _secondSubscriptionNameKey = "webjobstestsubscription2";
+        protected const string SecondSubscriptionNameKey = "%" + _secondSubscriptionNameKey  + "%";
+
+        private const string _secondaryNamespaceQueueKey = "webjobtestsecondarynamespacequeue";
+        protected const string SecondaryNamespaceQueueNameKey = "%" + _secondaryNamespaceQueueKey + "%";
+
+        // the connection key shouldn't use the % because the value specified is assumed to be a pointer
+        protected const string SecondaryConnectionStringKey = "webjobtestsecondaryconnection";
+
+        protected const int SBTimeoutMills = 120 * 1000;
+        protected const int DrainWaitTimeoutMills = 120 * 1000;
+        protected const int DrainSleepMills = 5 * 1000;
+        internal const int MaxAutoRenewDurationMin = 5;
+        internal static TimeSpan HostShutdownTimeout = TimeSpan.FromSeconds(120);
+
+        protected static QueueScope _firstQueueScope;
+        protected static QueueScope _secondaryNamespaceQueueScope;
+        private QueueScope _secondQueueScope;
+        private QueueScope _thirdQueueScope;
+        protected static TopicScope _topicScope;
+
+        private readonly bool _isSession;
+
+        protected WebJobsServiceBusTestBase(bool isSession)
+        {
+            _isSession = isSession;
+        }
+
+        /// <summary>
+        ///   Performs the tasks needed to initialize the test fixture.  This
+        ///   method runs once for the entire fixture, prior to running any tests.
+        /// </summary>
+        ///
+        [SetUp]
+        public async Task FixtureSetUp()
+        {
+            _firstQueueScope = await CreateWithQueue(enablePartitioning: false, enableSession: _isSession);
+            _secondQueueScope = await CreateWithQueue(enablePartitioning: false, enableSession: _isSession);
+            _thirdQueueScope = await CreateWithQueue(enablePartitioning: false, enableSession: _isSession);
+            _topicScope = await CreateWithTopic(
+                enablePartitioning: false,
+                enableSession: _isSession,
+                topicSubscriptions: new string[] { "sub1", "sub2" });
+            _secondaryNamespaceQueueScope = await CreateWithQueue(
```

Comment:
Out of curiosity, what's the benefit of using two namespaces rather than just two queues/topics/whatever under one namespace?

Reply:
The SB function extension tests require it.

Reply:
But why?   Is that a remnant of "we're just moving the existing stuff with minimal changes for now?"   or  is there some benefit that I can't think of for doing so?

Reply:
One of the tests explicitly tests the scenario of working across accounts. I think it is too early to say that we can disregard this case as not adding enough value. We can perhaps re-evaluate as we do the migration.